bin/activate.bat

```diff
@@ -1,78 +1,44 @@
+    set CONDA_NEW_ENV=root
+:skipmissingarg
 
-REM Search through paths in CONDA_ENVS_PATH
-REM First match will be the one used
+REM Use conda itself to figure things out
 
-for %%F in ("%CONDA_ENVS_PATH:;=" "%") do (
-    if exist "%%~F\%CONDA_NEW_NAME%\conda-meta" (
-       set CONDA_NEW_PATH=%%~F\%CONDA_NEW_NAME%
-       goto found_env
-    )
-)
+SET "CONDA_EXE=%~dp0\conda.bat"
 
-if exist "%CONDA_NEW_NAME%\conda-meta" (
-    set CONDA_NEW_PATH=%CONDA_NEW_NAME%
-    ) else (
-    echo No environment named "%CONDA_NEW_NAME%" exists in %CONDA_ENVS_PATH%, or is not a valid conda installation directory.
-    set CONDA_NEW_NAME=
-    set CONDA_NEW_PATH=
+REM TODO: will this work if Conda root env is not on PATH?
+
+REM Run secret conda ..checkenv command
+call "%CONDA_EXE%" ..checkenv %CONDA_NEW_ENV%
+REM EQU 0 means 0 or above on Windows ;(
+if %ERRORLEVEL% GTR 0 (
     exit /b 1
 )
 
-:found_env
-
-for /F %%i in ("%CONDA_NEW_PATH%") do set CONDA_NEW_NAME=%%~ni
-
 REM Deactivate a previous activation if it is live
-if "%CONDA_DEFAULT_ENV%" == "" goto skipdeactivate
-    REM This search/replace removes the previous env from the path
-    echo Deactivating environment "%CONDA_DEFAULT_ENV%"...
-
-    REM Run any deactivate scripts
-    if not exist "%CONDA_DEFAULT_ENV%\etc\conda\deactivate.d" goto nodeactivate
-        pushd "%CONDA_DEFAULT_ENV%\etc\conda\deactivate.d"
-        for %%g in (*.bat) do call "%%g"
-        popd
-    :nodeactivate
-
-    set CONDACTIVATE_PATH="%CONDA_DEFAULT_ENV%";"%CONDA_DEFAULT_ENV%\Scripts";"%CONDA_DEFAULT_ENV%\Library\bin"
-    call set PATH=%%PATH:%CONDACTIVATE_PATH%=%%
-    set CONDA_DEFAULT_ENV=
-    set CONDACTIVATE_PATH=
-    set PROMPT=%CONDA_OLD_PROMPT%
-    set CONDA_OLD_PROMPT=
-:skipdeactivate
-
-set CONDA_DEFAULT_ENV=%CONDA_NEW_PATH%
-echo Activating environment "%CONDA_DEFAULT_ENV%"...
-set PATH="%CONDA_DEFAULT_ENV%";"%CONDA_DEFAULT_ENV%\Scripts";"%CONDA_DEFAULT_ENV%\Library\bin";%PATH%
-IF "%CONDA_NEW_NAME%"=="" (
-   set PROMPT=$P$G
-   REM Clear CONDA_DEFAULT_ENV so that this is truly a "root" environment, not an environment pointed at root
-   set CONDA_DEFAULT_ENV=
-   ) ELSE (
-   set PROMPT=[%CONDA_NEW_NAME%] $P$G
+FOR /F "delims=" %%i IN ('"%CONDA_EXE%" ..deactivate') DO set PATH=%%i
+if %ERRORLEVEL% GTR 0 (
+exit /b 1
 )
-set CONDA_NEW_NAME=
-set CONDA_NEW_PATH=
-
-REM Run any activate scripts
-if not exist "%CONDA_DEFAULT_ENV%\etc\conda\activate.d" goto noactivate
-    pushd "%CONDA_DEFAULT_ENV%\etc\conda\activate.d"
-    for %%g in (*.bat) do call "%%g"
-    popd
-:noactivate
+REM Activate the new environment
+FOR /F "delims=" %%i IN ('"%CONDA_EXE%" ..activate %CONDA_NEW_ENV%') DO set PATH=%%i
+if %ERRORLEVEL% GTR 0 (
+exit /b 1
+)
+for /F %%C IN ('"%CONDA_EXE%" ..changeps1') DO set CHANGEPS1=%%C
+if "%CHANGEPS1%" == "1" set PROMPT=[%CONDA_NEW_ENV%] $P$G
+if %ERRORLEVEL% GTR 0 (
+exit /b 1
+)
+endlocal & set PROMPT=%PROMPT%& set PATH=%PATH%& set CONDA_DEFAULT_ENV=%CONDA_NEW_ENV%
```

Comment:
The variable name `CONDA_DEFAULT_ENV` is now inconsistent with the code in https://github.com/conda/conda/pull/1727, where you changed it into `CONDA_ACTIVE_ENV`. As I said elsewhere, I vote for the new name.